src/expire.c

```diff
@@ -521,6 +580,50 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
         return;
     }
 
+    /* NX option is set, check current expiry */
+    if (flag & EXPIRE_NX) {
+        current_expire = getExpire(c->db, key);
+        if (current_expire != -1) {
+            addReply(c,shared.czero);
+            return;
+        }
+    }
+
+    /* XX option is set, check current expiry */
+    if (flag & EXPIRE_XX) {
+        current_expire = getExpire(c->db, key);
```

Comment:
since XX and GT/LT can come together, we may be doing `getExpire` twice.
i wanna suggest a small refactoring (conceptually, not in LOC), and do something like this:
```c
    if (flag) {
        current_expire = getExpire(c->db, key);
        if (flag & EXPIRE_XX) {
...
        }
        if (flag & EXPIRE_NX) {
...
        }
        if (flag & EXPIRE_GT) {
...
        }
        if (flag & EXPIRE_LT) {
...
        }
    }